### PR TITLE
[RFC] vim-patch.sh improvements  [ci skip]

### PR DESCRIPTION
**Pull request by blueyed**
IDEAS:

- [ ] include (git commit) summary line with `-l`, and `-L` (?).  The latter is meant for external parsing, so should be handled with care, but the former updates Vim sources always.  Could be done later still - not that trivial.

- [ ] have a mode to report missing patches for a given file, which might be based on Vim (with optional ".vim-src" prefix) or Neovim.  This should be distinct, since it is non-trivial to e.g. adjust the path in `$@` when passing it to git-log.

### CODE REVIEW

File: scripts/vim-patch.sh [hunk 1]
```diff
@@ -107,9 +107,12 @@ commit_message() {
   fi
 }
 
-find_git_remote() {
-  git remote -v \
-    | awk '$2 ~ /github.com[:\/]neovim\/neovim/ && $3 == "(fetch)" {print $1; exit}'
+set_git_remote() {
+  git_remote=$(git remote -v \
```
Comment (justinmk) [inline scripts/vim-patch.sh]: this works on `local` variables?!

Comment (blueyed) [reply scripts/vim-patch.sh]: > this works on local variables?!

If the call is in the local scope, yes.

Will change it to echo "origin" then.

**Comment by blueyed** [comment]
Pushed an update to pass through git-log options with `-l` / `-L`.

**Comment by blueyed** [comment]
I think updating sources should be done less often - at least e.g. with `-L`.
What do you think in this regard?
It could for example do it automatically with `-P` etc when the given Vim revision is not found only, but should do it with `-l` of course still.

While this takes some time in general, it is also more noisy and might even cause unexpected issues when you are editing/comparing a file from below `.vim-src`, and it gets updated due to use of `-L`, which was only meant to list patches really.

**Comment by blueyed** [comment]
/cc @janlazo @erw7 
I think this is quite useful.  Any feedback?

**Review by justinmk — APPROVED** [review]

**Comment by janlazo** [comment]
@blueyed Can you add a sample line in the help message for this feature?

```sh
./scripts/vim-patch.sh -l -- src/testdir/runtest.vim
```
This filters the relevant patches for this file. Useful since I compare logs between Vim and Neovim on test files and I don't want to manually filter or keep a tab/split open for each log.

**Comment by janlazo** [comment]
It works with `-G` for filtering patches. Perhaps useful to note it in https://github.com/neovim/neovim/wiki/Merging-patches-from-upstream-Vim.

**Pull request merged**